frontend/src/components/organisms/RegisterUserForm.tsx/RegisterUserForm.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { useState } from "react";
+
+import * as yup from "yup";
+import { useFormik } from "formik";
+import { EntityIdGenerator } from "../../atoms/idGenerator/EntityIdGenerator";
+import { Button, Grid, TextField, Typography } from "@material-ui/core";
+import { VerticalSpace } from "../../atoms/VerticalSpace";
+import Notification from "../Notification/Notification";
+
+const validationSchema = yup.object({
+  name: yup.string().required("To pole jest wymagane."),
+  surname: yup.string().required("To pole jest wymagane."),
+  email: yup
+    .string()
+    .email("Niepoprawny adres e-mail.")
+    .required("To pole jest wymagane."),
+  phone: yup
+    .string()
+    .min(9, "Wprowadź przynajmniej 9 znaków.")
+    .required("To pole jest wymagane."),
+  password: yup
+    .string()
+    .min(5, "Wprowadź przynajmniej 5 znaków.")
+    .required("To pole jest wymagane."),
+  passwordConfirmation: yup
+    .string()
+    .oneOf([yup.ref("password"), null], "Hasła nie pasują do siebie")
+    .required("To pole jest wymagane."),
+});
+
+export function RegisterUserForm(props: { onDrawerClose: () => void }) {
+  const [openAlert, setOpenAlert] = useState(false);
+  const [textAlert, setTextAlert] = useState("");
+
+  const onNotificationOpen = (errorMessage: string) => {
+    setTextAlert(errorMessage);
+    setOpenAlert(true);
+  };
+
+  const onNotificationClose = (
+    event?: React.SyntheticEvent,
+    reason?: string
+  ) => {
+    if (reason === "clickaway") {
+      return;
+    }
+    setOpenAlert(false);
+  };
+
+  const formik = useFormik({
+    initialValues: {
+      name: "",
+      surname: "",
+      email: "",
+      phone: "",
+      password: "",
+      passwordConfirmation: "",
+    },
+    validationSchema: validationSchema,
+    onSubmit: async (values) => {
+      try {
+        const playerId = EntityIdGenerator.generate();
+        //TODO add player here
```

Comment:
Ci tutaj zamierzasz zrobic?

Reply:
Jak będzie zrobiony backend do tego to tutaj wszystko z formularza pójdzie w zapytaniu do stworzonego endpointa :-)